### PR TITLE
feat(SBE-538): client library

### PR DESCRIPTION
**Pull request by radekkrejcirik01**
**YouTrack task**
[SBE-538](https://knowledge.glcharge.com/issue/SBE-538)

**Adding:**
- [x] Feature
- [ ] Bug fix
- [ ] Test coverage

**Why is this change required?**
- We are generating OCPI client library for our services as well for open source purposes

**What changed?**
- Generated client library from all modules specifications using `opai-codegen`

### CODE REVIEW

File: model_authorization_info_info.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+type AuthorizationInfoInfo struct {
+	Language string `json:"language"`
+	Text string `json:"text"`
+}
+
+// NewAuthorizationInfoInfo instantiates a new AuthorizationInfoInfo object
+// This constructor will assign default values to properties that have it defined,
+// and makes sure properties required by API are set, but the set of arguments
+// will change when the set of required properties is changed
+func NewAuthorizationInfoInfo(language string, text string) *AuthorizationInfoInfo {
```
Comment (xBlaz3kx) [inline model_authorization_info_info.go:30]: InfoInfo is not a good naming convention.

File: model_cdr_cdr_token.go [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+import (
+	"encoding/json"
+)
+
+// checks if the CdrCdrToken type satisfies the MappedNullable interface at compile time
+var _ MappedNullable = &CdrCdrToken{}
+
+// CdrCdrToken struct for CdrCdrToken
+type CdrCdrToken struct {
```
Comment (xBlaz3kx) [inline model_cdr_cdr_token.go:21]: Naming here isn't optimal

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,6 @@
+module github.com/GIT_USER_ID/GIT_REPO_ID
```
Comment (xBlaz3kx) [inline go.mod:1]: This won't really work

File: api_default.go [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+	"net/url"
+)
+
+
+// DefaultApiService DefaultApi service
+type DefaultApiService service
+
+type ApiOcpi221DetailsGetRequest struct {
+	ctx context.Context
+	ApiService *DefaultApiService
```
Comment (xBlaz3kx) [inline api_default.go:27]: There's only one Service - for versions. Other API's aren't included for some reason

File: api/openapi.yaml [hunk 5]
```diff
@@ -0,0 +1,189 @@
+openapi: 3.0.3
```
Comment (xBlaz3kx) [inline api/openapi.yaml:1]: Why is this file included?

**Review by xBlaz3kx — CHANGES_REQUESTED** [review]

File: api_default.go (hunk 4 above)
Comment (radekkrejcirik01) [reply api_default.go:505]: fixed ✅

File: model_authorization_info_info.go (hunk 1 above)
Comment (radekkrejcirik01) [reply model_authorization_info_info.go]: resolved ✅

File: model_cdr_cdr_token.go (hunk 2 above)
Comment (radekkrejcirik01) [reply model_cdr_cdr_token.go]: resolved as cdrBody, let me know if you have would have better idea 🤔

File: README.md [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+ - [Versions](docs/Versions.md)
+ - [VersionsData](docs/VersionsData.md)
+
+
+## Documentation For Authorization
+
+Endpoints do not require authorization.
+
+
+## Documentation for Utility Methods
```
Comment (xBlaz3kx) [inline README.md:232]: We can probably omit or move this.

File: README.md [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+ - [TokenResponse](docs/TokenResponse.md)
+ - [TokensResponse](docs/TokensResponse.md)
+ - [UnlockConnector](docs/UnlockConnector.md)
+ - [Versions](docs/Versions.md)
+ - [VersionsData](docs/VersionsData.md)
+
+
+## Documentation For Authorization
+
+Endpoints do not require authorization.
```
Comment (xBlaz3kx) [inline README.md:229]: This is not entirely true - most of the modules require Bearer Token authorization per OCPI.

File: README.md [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+```
+
+## Documentation for API Endpoints
+
+All URIs are relative to *http://localhost*
+
+```shell
+Class | Method | HTTP request | Description
+------------ | ------------- | ------------- | -------------
+*DefaultApi* | [**Ocpi221CredentialsDelete**](docs/DefaultApi.md#ocpi221credentialsdelete) | **Delete** /ocpi/2.2.1/credentials | Informs that credentials are invalid
```
Comment (xBlaz3kx) [inline README.md:91]: The classes should be each module, so we can nicely split this functionality

**Review by xBlaz3kx — CHANGES_REQUESTED** [review]

File: README.md (hunk 7 above)
Comment (radekkrejcirik01) [reply README.md]: I removed it, it was autogenerated

File: README.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+import OCPI "https://github.com/GLCharge/ocpi/<ocpi-version>"
+```
+
+## Documentation
+
+- [2.2.1](https://www.github.com/GLCharge/ocpi/2.1.1/api/)
+
+## Generation
+
+This code base, was generated using [opai-codegen](https://github.com/deepmap/oapi-codegen).
```
Comment (xBlaz3kx) [inline README.md:23]: Misspelt - `oapi-codegen`

**Review by xBlaz3kx — APPROVED** [review]

File: README.md (hunk 9 above)
Comment (radekkrejcirik01) [reply README.md]: fixed ✅

File: README.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+This repository contains client libraries for Open Charge Points Interfaces modules.
+
+Every version has OpenAPI specification that is used to generate the client using `opai-codegen` generator.
+OpenApi specifications contain also documentation. For more detailed documentation about the modules and APIs please follow
+the official [OCPI documentation](https://evroaming.org/app/uploads/2021/11/OCPI-2.2.1.pdf)
+
+## Installation
+
+```shell
```
[the remark this answers is not in the document]
Comment (radekkrejcirik01) [reply README.md:13]: when I use the `go` the Goland throws errors as if it would be `go` file 🤷‍♂️, I would keep the shell or remove it without specifying

[the remark this answers is not in the document]
Comment (radekkrejcirik01) [reply README.md]: used `go` ✅
